### PR TITLE
QPD-714: Updated the graphql url validator

### PR DESCRIPTION
**Pull request by tushar-quartic**
URL tested:

1. https://nginx/
2. http://<ip>/
3. https://<ip>/
4. Standard urls like (stag.quartic.ai and qa.quartic.ai)
5. Also tested url with invaild schema like - (htt://nginx)

### CODE REVIEW

File: quartic_sdk/graphql_client.py [hunk 1]
```diff
@@ -5,7 +5,10 @@
 import coloredlogs
 from quartic_sdk._version import __version__
 from typing import Optional, Union
-import validators
+from urllib.parse import urlparse
+import re
+
+SCHEMA_REGEX = re.compile(r"(?:(?:https?|ftp)://)")
```
Comment (shivinquartic) [inline quartic_sdk/graphql_client.py:11]: ftp is not required.

File: quartic_sdk/graphql_client.py [hunk 2]
```diff
@@ -43,7 +47,7 @@ class initialisation
         self.logger = logging.getLogger()
         coloredlogs.install(level='DEBUG', logger=self.logger)
 
-    @staticmethod
+    @ staticmethod
```
Comment (mprasoonquartic) [inline quartic_sdk/graphql_client.py:50]: space in staticmethod?

Comment (shivinquartic) [inline quartic_sdk/graphql_client.py:50]: Why this space?

Comment (tushar-quartic) [reply quartic_sdk/graphql_client.py]: Removed

Comment (tushar-quartic) [reply quartic_sdk/graphql_client.py]: Removed

File: quartic_sdk/graphql_client.py (hunk 1 above)
Comment (tushar-quartic) [reply quartic_sdk/graphql_client.py:11]: Removing

**Review by DineshQuartic — APPROVED** [review]

File: quartic_sdk/graphql_client.py (hunk 1 above)
Comment (tushar-quartic) [reply quartic_sdk/graphql_client.py]: Done

File: quartic_sdk/graphql_client.py [hunk 3]
```diff
@@ -80,17 +85,22 @@ def _get_graphql_url(self) -> str:
         Generates the graphql endpoint.
         """
         __graphql_url = f'{self.url}/graphql/'
-        if not validators.url(__graphql_url):
-            raise AttributeError(f'url entered is not correct = {self.url}')
-        return f'{self.url}/graphql/'
+        result = urlparse(__graphql_url)
+        if result.scheme and not SCHEMA_REGEX.match(__graphql_url):
+            raise AttributeError(
+                f'Invalid URL {self.url}: No schema supplied. Perhaps you meant http://...?')
```
Comment (scriperdjq) [inline quartic_sdk/graphql_client.py:91]: we throw exception is the url is https?

Comment (shivinquartic) [reply quartic_sdk/graphql_client.py:91]: Nope. @tushar-quartic I think you should update this message and validate it with Jacob once. We allow both http and https.

Comment (tushar-quartic) [reply quartic_sdk/graphql_client.py:91]: Will update it and validate from jacob

Comment (tushar-quartic) [reply quartic_sdk/graphql_client.py]: Updated the error message as per reviewed by Jacob

**Review by shivinquartic — APPROVED** [review]
lgtm

**Review by scriperdjq — APPROVED** [review]

**Pull request merged**